packages/appsync-modelgen-plugin/src/visitors/appsync-java-visitor.ts

```diff
@@ -843,10 +890,27 @@ export class AppSyncModelJavaVisitor<
   /**
    * Generate the name of the step builder interface
    * @param nextFieldName: string
+   * @param types: string - set of types for all fields on model
    * @returns string
    */
-  private getStepInterfaceName(nextFieldName: string): string {
-    return `${pascalCase(nextFieldName)}Step`;
+  private getStepInterfaceName(nextFieldName: string, types: Set<string>): string {
+    const pascalCaseFieldName = pascalCase(nextFieldName);
+    const stepInterfaceName = `${pascalCaseFieldName}Step`;
+
+    if (types.has(stepInterfaceName)) {
+      return `${pascalCaseFieldName}BuildStep`;
```

Comment:
This could still collide if for a model `A`, there are `AStep` and `ABuildStep` in the fields. But it should be a rare case. The implementation is enough to fix the issue.

Reply:
I agree that this case would be so rare that this fix should be sufficient.